ui/guest/src/react/Field.tsx

```diff
@@ -41,10 +42,18 @@ export const Field = forwardRef<any, FieldProps>(function <P = {}>(props: FieldP
     ref
   });
   const Component = component as ComponentType<P>;
+
+  const contentTypeId = model.craftercms.contentTypeId;
+  const contentType = contentController.getCachedContentType(contentTypeId);
+  const field = contentType?.fields[fieldId];
+
   const passDownProps = {
     ...other,
     ...ice,
     ...componentProps,
+    'data-craftercms-model-id': model.craftercms.id,
+    ...(field &&
+      ['node-selector', 'checkbox-group', 'repeat'].includes(field.type) && { 'data-craftercms-type': 'collection' }),
```

Comment:
Why would the checkbox group require the attribute?
```suggestion
      ['node-selector', 'repeat'].includes(field.type) && { 'data-craftercms-type': 'collection' }),
```

Move this whole expression out of the primitive initialization. We don't need to inline everything here.